pkg/controller/datavolume-controller.go

```diff
@@ -808,6 +806,9 @@ func (r *DatavolumeReconciler) ensureExtendedToken(pvc *corev1.PersistentVolumeC
 func (r *DatavolumeReconciler) selectCloneStrategy(datavolume *cdiv1.DataVolume, pvcSpec *corev1.PersistentVolumeClaimSpec) (cloneStrategy, error) {
 	preferredCloneStrategy, err := r.getCloneStrategy(datavolume)
 	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return NoClone, nil
```

Comment:
maybe it would be better if `getCloneStrategy()` handles this case?

Reply:
what should be returned in that case? preferredCloneStrategy=nil and err=nil? I would still need to add here a check if preferredCloneStrategy=nil and return NoClone. or create a new CDICloneStrategy-> cdiv1.CloneStrategyNoClone? I would still need to add here a check of if preferredCloneStrategy=cdiv1.CloneStrategyNoClone and return NoClone?
I dont mind doing either but doesnt seem any cleaner or simpler then this

Reply:
If this behaves correctly then I am ok with the rest.

Reply:
@brybacki I think it behaves as expected: If the source PVC doesn't exist but the target is already populated, we are just updating the DV without triggering any errors. However, if the source PVC doesn't exist and the target is not populated, it'll be handled by `validateCloneAndSourcePVC`, which will exit without errors to retry once the source is created:

https://github.com/kubevirt/containerized-data-importer/blob/083ee010753fea2d2d7e004e8afd48ea3bd90fe4/pkg/controller/datavolume-controller.go#L673-L682

Reply:
ok